metadata-ingestion/src/datahub/ingestion/source/schema_inference/json.py

```diff
@@ -29,9 +29,24 @@
 
 
 class JsonInferrer(SchemaInferenceBase):
-    def infer_schema(self, file: IO[bytes]) -> List[SchemaField]:
+    def merge_new_dictionary(self, complete_dict, new_dict):
+        for key in new_dict.keys():
+            if key not in complete_dict.keys():
+                complete_dict[key] = new_dict[key]
+
+        return complete_dict
+
 
-        datastore = ujson.load(file)
+    def parse_json(self, file):
+        complete_datastore = {}
+        for line in file:
+            datastore = ujson.loads(line.decode('utf8'))
+            complete_datastore = self.merge_new_dictionary(complete_datastore, datastore)
```

Comment:
It is possible that a line of JSON in a file is an array/list as opposed to being an object/dict. Or for that matter just a single string, etc.

It looks like we don't have to do the work to handle all these cases, however. The `construct_schema` function called below on line 54 is already doing the necessary work. So all we need to do is pass it a list of parsed JSON lines. Right now it is being passed a list with a single element, but it handles merging/inferring schema from multiple elements in the list.

You can take a look at the [source code](https://vscode.dev/github/Affirm/datahub/blob/fed6bec579be154b83d3e07930f5ac69cfc092c6/metadata-ingestion/src/datahub/ingestion/source/schema_inference/object.py#L85) for `contruct_schema`.